packages/manager/apps/container/src/core/onboarding/useOnboarding.js

```diff
@@ -9,32 +11,55 @@ export const useOnboarding = () => {
   const reketInstance = useReket();
   const preferenceKey = 'NAV_RESHUFFLE_ONBOARDING';
 
+  const getOnboardingFromLocalStorage = () => {
+    return window.localStorage.getItem(preferenceKey);
+  };
+
+  const setOnboardingToLocalStorage = (value) => {
+    return new Promise((resolve) => {
+      window.localStorage.setItem(preferenceKey, JSON.stringify(value));
+      resolve(null);
+    });
+  };
+
   const createPreference = () => {
     const value = { status: ONBOARDING_STATUS_ENUM.DISPLAYED };
 
-    return reketInstance
-      .post(`/me/preferences/manager`, {
-        key: preferenceKey,
-        value: JSON.stringify(value),
-      })
-      .then(() => {
-        return value;
-      });
+    const createPromise = isBetaForced()
+      ? setOnboardingToLocalStorage(value)
+      : reketInstance.post(`/me/preferences/manager`, {
+          key: preferenceKey,
+          value: JSON.stringify(value),
+        });
+
+    return createPromise.then(() => value);
   };
 
   const updatePreference = (value) => {
-    return reketInstance
-      .put(`/me/preferences/manager/${preferenceKey}`, {
-        value: JSON.stringify(value),
-      })
-      .then(() => {
-        return value;
-      });
+    const updatePromise = isBetaForced()
+      ? setOnboardingToLocalStorage(value)
```

Comment:
i think setOnboardingToLocalStorage should not return a promise since the action is synchronous,
you could just do :
```js
const updatePromise = isBetaForced() ?
  Promise.resolve(() => setOnboardingToLocalStorage(value)) :
  reketInstance.put(...);
```

same for the createPromise